dashboard/app/dsl/level_group_dsl.rb

```diff
@@ -27,6 +28,19 @@ def page
   end
 
   def level(name)
+    # Ensure level name hasn't already been used.
+    if @level_names.include? name
+      raise "Don't use the same level twice in a LevelGroup (#{name})."
+    end
+    @level_names << name
+
+    # Ensure level is appropriate type.
+    level = Level.find_by_name(name)
```

Comment:
Yes, I think this is a reasonable expectation and lets us do stronger validation.